### PR TITLE
Enable loading of restarted postprocessor values into a receiver

### LINKED ISSUE #25414: Support loading restarted postprocessor values in a `Reporter`
## Reason
It is desired to have something like this in a checkpointed run:

```
[Postprocessors/value]
  type = SomePostprocessor
[]
```

and then to support loading this value in restart via:

```
[Postprocessors/value]
  type = Receiver
[]
```

## Design
Support the above in test it. 

## Impact
New feature.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #25414

### CODE REVIEW

**Comment by loganharbour** [comment]
@lindsayad this is in support of transferring the eigenvalue for restart. It's the simple solution for now

File: framework/include/reporters/ReporterState.h [hunk 1]
```diff
@@ -190,19 +197,36 @@ ReporterState<T>::copyValuesBack()
...
 
   // If the current container is undersized, expand it to fit the loaded data
-  if (this->get().size() < size)
-    this->set().resize(size);
+  if (values.size() < size)
+    values.resize(size);
 
   // Load each entry of the list directly into the storage
-  for (auto & val : this->set())
+  // Because we don't shrink the container if the stored size is smaller than
```
Comment (lindsayad) [inline framework/include/reporters/ReporterState.h:228]: why don't we shrink?

File: test/tests/restart/receiver/tests [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  design = "Restartable.md Receiver.md"
+  issues = "#25414"
+
+  [checkpoint]
+    type = 'CSVDiff'
+    input = 'receiver_initial.i'
+    csvdiff = 'receiver_initial_out.csv'
+    recover = false
+
+    requirement = "The system shall support the checkpointed output of postprocessors"
```
Comment (lindsayad) [inline test/tests/restart/receiver/tests:11]:
```suggestion
    requirement = "The system shall support the checkpointed output of postprocessors."
```

File: test/tests/restart/receiver/tests [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  []
+  [restart]
+    type = 'CSVDiff'
+    input = 'receiver_restart.i'
+    csvdiff = 'receiver_restart_out.csv'
+    recover = false
+    prereq = checkpoint
+
+    requirement = "The system shall support loading a postprocessor into another postprocessor during restart"
```
Comment (lindsayad) [inline test/tests/restart/receiver/tests:20]:
```suggestion
    requirement = "The system shall support loading a postprocessor into another postprocessor during restart."
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Comment by lindsayad** [comment]
> @lindsayad this is in support of transferring the eigenvalue for restart. It's the simple solution for now

Did we need https://github.com/idaholab/moose/pull/24510 for this capability?

**Comment by loganharbour** [comment]
> > @lindsayad this is in support of transferring the eigenvalue for restart. It's the simple solution for now
> 
> Did we need #24510 for this capability?

Not for transferring the eigenvalue like this. #24510 is needed to pull system vectors from eigenvalue -> non eigenvalue.

This is the really quick, lazy way to get the scalar eigenvalue from one solve to another.

In time, we would prefer to not do it like this and access the underlying restartable data in an automated way. Which #24510 does enable.

File: framework/include/reporters/ReporterState.h (hunk 1 above)
Comment (loganharbour) [reply framework/include/reporters/ReporterState.h:228]: Example: user requests old and older postprocessor values in another object. This means the storage goes from size 1 to size 3. We then go to load from restart. Well, the restart only has a current value. Shrink would mean to undo the user request for the user requesting the old and older values. Instead, we want to keep the user's request for old and older, pull in current from restart, and then copy current from restart back to old and older later.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**